layers/ip6.go

```diff
@@ -34,6 +75,7 @@ type IPv6 struct {
 	HopLimit     uint8
 	SrcIP        net.IP
 	DstIP        net.IP
+	RoutingDstIP net.IP
```

Comment:
It is possible to simply not use the RoutingDst field in IPv6 struct for decoding, therefore avoiding the whole issue of changing PacketBuilder with LastNetworkLayer(). That field is only essential when serializing to be able to compute the checksum correctly. When decoding it is redundant.